<launch>
    <arg name="gaze_topic" default="/gaze_publisher/gaze"/>
    <arg name="roi_topic" default="/carla/ego_vehicle/camera/rgb/front/roi"/>
    <arg name="image_topic" default="/carla/ego_vehicle/camera/rgb/front/image_color"/>
    <arg name="visualize" default="True"/>
    <arg name="visualize_debug" default="False"/>
    <arg name="config_path" default="$(find awareness_detector)/config/awareness_model_config.yaml"/>

    <node name="awareness_detection" pkg="awareness_detector" type="awareness_model.py" respawn="false" output="screen">
        <param name="gaze_topic" value="$(arg gaze_topic)"/>
        <param name="roi_topic" value="$(arg roi_topic)"/>
        <param name="image_topic" value="$(arg image_topic)"/>
        <param name="visualize" value="$(arg visualize)"/>
        <param name="visualize_debug" value="$(arg visualize_debug)"/>
        <rosparam file="$(arg config_path)" command="load"/>
    </node>
</launch>
